### DISCUSSION TITLE
Google oauth redirects to localhost on production

### DISCUSSION THREAD

**Discussion by umang-goyal** [supabase/supabase#25756] [Questions] [answered]
I deployed my nextjs app to vercel on `https://dailyclimb.vercel.app/`, when im trying to "signup with google" ( `https://dailyclimb.vercel.app/signup` )  it redirects me to `localhost:3000?code=xyz`. 

Not sure what am i missing. I have no where hardcoded localhost:3000 in my code. 
Function I am using google ouath
```
const handleLogin = async () => {
    const supabase = createSBaseBrowserClient();
    await supabase.auth.signInWithOAuth({
      provider: 'google',
      options: {
        scopes: 'https://www.googleapis.com/auth/calendar',
        redirectTo: `${location.origin}/auth/callback?next=/onboarding`,
      }
    })
  }
```

**Comment by umang-goyal** [comment, marked as answer]
After going to production you have to update the site url and add your production site url in redirection urls. 
![image](https://github.com/supabase/supabase/assets/42673134/c1c63811-ffeb-4032-89ec-1b021f210a28)

**Reply by pablodz** [reply]
https://github.com/orgs/supabase/discussions/1514#discussioncomment-10062334

**Comment by AndryRoby** [comment]
Glad you got there. Worth adding for anyone else landing here from search: updating Site URL alone isn't always enough, because Supabase checks `redirectTo` against the separate Redirect URLs allow-list too, and it's an exact-match/wildcard check, not "same domain counts". So `https://dailyclimab.vercel.app/auth/callback` needs to be its own allow-list entry, not just the bare origin. And if you ever deploy preview branches, you'll want a wildcard entry as well, since each preview gets a different `*.vercel.app` subdomain that won't match one hardcoded URL.

I built a free client-side checker for this exact pattern (Site URL vs. allow-list vs. callback route mismatches), since it trips up so many Next.js + Vercel + Supabase setups: https://arling.sk/supabase-redirect-doctor/
